src/hotspot/cpu/x86/vm_version_x86.cpp

```diff
@@ -895,8 +895,9 @@ void VM_Version::get_processor_features() {
     }
   }
   if (FLAG_IS_DEFAULT(UseAVX)) {
-    // Don't use AVX-512 on older Skylakes unless explicitly requested.
-    if (use_avx_limit > 2 && is_intel_skylake() && _stepping < 5) {
+    // Don't use AVX-512 on Skylake (or the related Cascade Lake) CPUs unless explicitly
+    // requested - these instructions can cause performance issues on these processors.
+    if (use_avx_limit > 2 && is_intel_skylake()) {
```

Comment:
Maybe `is_intel_skylake` needs to be changed to `is_cpu_model_intel_skylake`? It will make clear that all CPUs based on Skylake model are excluded.

Reply:
I agree it's not necessarily a perfect name, but I haven't changed its behaviour so I figured I can avoid making my change any bigger than necessary. The intention of this particular usage is clarified in my comment. It's used in [another place](https://github.com/openjdk/jdk/blob/d77e5680af382f8215b5b1f9cd4754056bccc9e5/src/hotspot/cpu/x86/vm_version_x86.hpp#L1083) too, where it's evidently understood to include Cascade lake since the comment mentions Ice lake + (Cascade lake successor).